### PR TITLE
Pipeline integration

### LINKED ISSUE #32: Refactor pipelines
HuggingFace transformers' pipelinies underwent in a major major refactor in [v4.11.0](https://github.com/huggingface/transformers/releases/tag/v4.11.0). Thanks to the new design, we can refactor our pipeline factory and existing pipelines as well as possibly wrapping the pipeline as a Registrable class. However, [allennlp's latest release](https://pypi.org/project/allennlp/2.7.0/) does not have that yet although their master branch has. Therefore, we may need to wait till their new release (I expect it won't take long) or use [that commit](https://github.com/allenai/allennlp/commit/a63e28c24d091fb371011f644fec3ebd29bfbb7e) to prevent a dependency mismatch.

### PR DESCRIPTION
**Pull request by devrimcavusoglu**
Closes #32

### CODE REVIEW

File: trapper/metrics/input_handlers/token_classification_input_handler.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ def preprocess(self, eval_pred: EvalPrediction) -> EvalPrediction:
             actual_labels.append(actual_label)
 
         processed_eval_pred = EvalPrediction(
-            predictions=predictions, label_ids=references
+            predictions=np.array(actual_predictions),
+            label_ids=np.array(actual_labels),
```
Comment (Sophylax) [inline trapper/metrics/input_handlers/token_classification_input_handler.py:52]: Is this a bug fix?

**Review by Sophylax — COMMENTED** [review]
Apart from one needed clarification, this seems great.

File: trapper/metrics/input_handlers/token_classification_input_handler.py (hunk 1 above)
Comment (devrimcavusoglu) [reply trapper/metrics/input_handlers/token_classification_input_handler.py:52]: Yes, the former version was just ignoring the preprocess step.

**Review by Sophylax — APPROVED** [review]
LGTM!

**Pull request merged**